ops/services/app_functions/report_stream_batched_publisher/functions/package.json

```diff
@@ -0,0 +1,36 @@
+{
+  "name": "report-stream-batched-publisher",
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
+    "func": "func",
+    "clean": "rimraf build",
+    "build:compile": "tsc",
+    "build:production": "jest --coverage && npm run prestart && npm prune --production && npm run build:zip",
+    "prebuild:zip": "mkdir -m700 -p build",
+    "build:zip": "zip -r build/batched-rs-publisher.zip . -x \"*.zip\" -x \".terraform\" -x \"terraform.*\" -x \"*.tf\" -x \"local.settings.json\" -x \"*.map\"",
```

Comment:
There is a really cool package you can use to do this https://stuk.github.io/jszip/ :)

Reply:
😬 

So, we _started_ with https://github.com/ffflorian/jszip-cli - but it's not v actively maintained and I ran into a huge issue where directories that were too-deeply nested got excluded from the zip regardless of configuration and w/o any warning. This of course broke the build on some sub-dependencies

I switched to using system-native zip & mkdir (instead of jszip and mkdirp) to avoid this problem... of course it's no longer OS-agnostic, but 99% of this team is on Mac OS or Linux anyway. I'd like to keep this as-is for now and deal with it later if/when we have a Windows user who gets stuck

Reply:
It was a bad joke, I know how much of a problem it caused you.

Reply:
lolol 😭